src/Microsoft.Framework.Configuration/ConfigurationBase.cs

```diff
@@ -86,12 +86,20 @@ public IConfigurationSection GetSection(string key)
         {
             if (string.IsNullOrEmpty(key))
             {
-                throw new InvalidOperationException(Resources.Error_EmptyKey);
+                throw new ArgumentException(Resources.Error_EmptyKey);
             }
 
-            return new ConfigurationSection(Sources, GetPrefix() + key);
+            return new ConfigurationSection(Sources, GetPath(), key);
         }
 
-        protected abstract string GetPrefix();
+        private string GetPath()
```

Comment:
I think we have had a misunderstanding. It seems that `GetPrefix()` would still be the right name for this private method that appends the delimiter if needed.